text/0043-block-metadata.md

```diff
@@ -0,0 +1,193 @@
+`struct BlockMetadata` to `mc_blockchain_types::BlockData`.
+
+## Protobuf schema
+
+```protobuf
+message BlockMetadataContents {
+    /// The Block ID.
+    BlockID id = 1;
+
+    /// Quorum set configuration at the time of externalization.
+    QuorumSet quorum_set = 2;
+
+    /// IAS report for the enclave which generated the signature.
+    VerificationReport verification_report = 3;
+}
+
+message BlockMetadata {
+    /// Metadata signed by the consensus node.
+    BlockMetadataContents contents = 1;
+
+    /// Message signing key.
+    Ed25519Public node_key = 2;
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    Ed25519Signature signature = 3;
+}
+
+// NB: Only addition is field #4.
+message ArchiveBlockV1 {
+    /// Block.
+    Block block = 1;
+
+    /// Contents of the block.
+    BlockContents block_contents = 2;
+
+    /// Block signature.
+    BlockSignature signature = 3;
+
+    /// Additional signed metadata about this block.
+    BlockMetadata metadata = 4;
+}
+```
+
+Since this is a protobuf-additive change, we don't need a new `ArchiveBlock`
+version/type.
+
+## Signing metadata
+The metadata signature will always differ from the block signature. The block
+signature is generated by the enclave using a private key that never leaves the
+enclave, while the metadata is signed by the "untrusted" code outside the
+enclave using the SCP message signing key, since the enclave does not have this
+metadata.
+
+The metadata signature is generated from the `Digestible` encoding of
+`BlockMetadataContents`, since the protobuf wire format is not a canonical
+encoding.
+
+## Signing key history
+MobileCoin Foundation will publish a mapping/list of node_id to a range of block
+indexes, which node operators and consumers can use to verify the metadata
+signatures.
+
+This mapping will contain multiple node_id:block_range entries, in the following
+TOML format:
+```toml
+[[node]]
+message_signing_pub_key = ""  # Pub key, as PEM-encoded data or a file path containing such data.
+first_block_index = X  # Index of the first block signed with this key.
+last_block_index = Y  # Optional index of the last block signed with this key.
+```
+
+## Historical AVRs
+With `BlockMetadata`, consumers of the blockchain no longer need a Watcher DB.
+Ledger DB will default to reading AVRs from `BlockMetadata`, and fall back to a
+lookup table containing AVRs for older blocks.
+
+MobileCoin Foundation will publish a bootstrap file defining this lookup table
+with historical consensus enclave AVRs, up to when consensus nodes publish
+`BlockMetadata` with their AVRs.
+
+The lookup data will be in the following TOML format:
+```toml
+[[node]]
+responder_id = ""  # Responder ID for the consensus node.
+avr = ""  # AVR for the node's enclave, includes pubkey.
+first_block_index = X  # Index of the first block signed with this enclave.
+last_block_index = Y  # Optional index of the last block signed with this enclave.
```

Comment:
Since these are historical, we should have a cut over time.  Should `last_block_index` be required here?

Reply:
> Since these are historical, we should have a cut over time. Should `last_block_index` be required here?

Good catch!

```suggestion
last_block_index = Y  # Index of the last block signed with this enclave.
```

> It probably wouldn't be strictly required, but it'd make bootstrapping dbs with it more of a pain/error prone since you'd have to somehow access the next record and be sure that's actually the next record.

Not necessarily. We just need to verify that the current block's index is within the range.